fract4d/c/fract4dc/calcs.cpp

```diff
@@ -47,24 +48,8 @@ namespace calcs {
         {
             cargs->site->interrupt();
             cargs->site->wait();
-
-            // cargs->site->start(cargs);
             cargs->site->start();
-
-            pthread_t tid;
-
-            /* create low-priority attribute block */
-            pthread_attr_t lowprio_attr;
-            //struct sched_param lowprio_param;
-            pthread_attr_init(&lowprio_attr);
-            //lowprio_param.sched_priority = sched_get_priority_min(SCHED_OTHER);
-            //pthread_attr_setschedparam(&lowprio_attr, &lowprio_param);
-
-            /* start the calculation thread */
-            pthread_create(&tid, &lowprio_attr, calculation_thread, (void *)cargs);
-            assert(tid != 0);
-
-            cargs->site->set_tid(tid);
+            cargs->site->set_thread(std::thread(calculation_thread, cargs));
```

Comment:
This feels like, for reliability, it shouldn't be called via the cargs pointer, IE this:
```
const auto &site = *cargs->site;
site.set_thread(std::thread{calculation_thread, cargs});
```

This might also help the compiler with the other 3 calls to functions on site to do the same for them as it gets rid of a double-indirection..
Otherwise this looks way better!

Reply:
good point! (site couldn't be const though)

Reply:
ah, fair enough.. my default is to start const and remove as necessary :)